test/test_dbc.cpp

```diff
@@ -49,4 +49,80 @@ TEST_CASE("Testing dbc file loading", "[fileio]") {
 		REQUIRE(parser->get_messages().front().signals == msg.signals);
 	}
 
-}
\ No newline at end of file
+}
+
+/*!
+ * \brief TEST_CASE
+ * Test negative values in offset, min, max
+ */
+TEST_CASE("Testing negative values") {
+  const auto* filename = std::tmpnam(NULL);
+
+  auto* file = std::fopen(filename, "w");
+  CHECK(file);
+
+  std::fputs(PRIMITIVE_DBC.c_str(), file);
+  std::fputs(R"(BO_ 234 MSG1: 8 Vector__XXX
+ SG_ Sig1 : 55|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig2 : 39|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig3 : 23|16@0- (10,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig4 : 7|16@0- (1,-10) [0|32767] "" Vector__XXX)", file);
+  std::fclose(file);
+
+  auto parser = libdbc::DbcParser();
+  parser.parse_file(std::string(filename));
+
+  REQUIRE(parser.get_messages().size() == 1);
+  REQUIRE(parser.get_messages().at(0).signals.size() == 4);
+  {

```

Comment:
One suggestion with these are to use the `SECTION` tags to indicate what is happening below. For instance, this one I would probably have:
`SECTION("Parsing first message with negative min and negative max")`
if that is what is happening

https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md

Reply:
Thanks I will check it out

Reply:
fixed in 8053bba7f104d7bab0abc3c794b8de05d59aa15f

Reply:
Thanks for the addition here! looks good!